

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>RNA Pipeline Quickstart &mdash; GEMTools 1.6 documentation</title>
    
    <link rel="stylesheet" href="static/haiku.css" type="text/css" />
    <link rel="stylesheet" href="static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/breathe.css" type="text/css" />
    <link rel="stylesheet" href="_static/print.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    '',
        VERSION:     '1.6',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="static/jquery.js"></script>
    <script type="text/javascript" src="static/underscore.js"></script>
    <script type="text/javascript" src="static/doctools.js"></script>
    <script type="text/javascript" src="_static/theme_extras.js"></script>
    <link rel="top" title="GEMTools 1.6 documentation" href="index.html" />
    <link rel="prev" title="Welcome to GEMTools’s documentation!" href="index.html" /> 
  </head>
  <body>
      <div class="header"><h1 class="heading"><a href="index.html">
          <span>GEMTools 1.6 documentation</span></a></h1>
        <h2 class="heading"><span>RNA Pipeline Quickstart</span></h2>
      </div>
      <div class="topnav">
      
        <p>
        «&#160;&#160;<a href="index.html">Welcome to GEMTools&#8217;s documentation!</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="index.html">Contents</a>
        </p>

      </div>
      <div class="content">
        
        
  <div class="section" id="rna-pipeline-quickstart">
<span id="rna-pipeline"></span><h1>RNA Pipeline Quickstart<a class="headerlink" href="#rna-pipeline-quickstart" title="Permalink to this headline">¶</a></h1>
<div class="section" id="download-and-install-the-gemtools">
<h2>1) Download and install the gemtools<a class="headerlink" href="#download-and-install-the-gemtools" title="Permalink to this headline">¶</a></h2>
<p>If that goes well, you will have a &#8220;gemtools&#8221; command line tool available.
Please check the <a class="reference external" href="http://gemtools.github.io/">GEMTools homepage</a> for download
and installation instructions.</p>
</div>
<div class="section" id="create-a-genome-index">
<h2>2) Create a genome index<a class="headerlink" href="#create-a-genome-index" title="Permalink to this headline">¶</a></h2>
<p>In order to run the pipeline for any given genome, you need to create a gem
index for that particular genome. The gem-indexer takes a single single fasta
file as input. Assume you have a genome.fa file, you can create the .gem index
by calling:</p>
<div class="highlight-python"><pre>gemtools index -i genome.fa -t 8</pre>
</div>
<p>Here <tt class="docutils literal"><span class="pre">-t</span> <span class="pre">8</span></tt> indicates to use 8 threads/cpus, put this to the number of cpus
available to you. The command will create 3 files just next to the fasta file:</p>
<div class="highlight-python"><pre>genome.gem  -- the gem index
genome.hash -- hash version of the genome. You probably don't need this
               and the creation can be avoided using --no-hash
genome.log  -- log file of the indexing process</pre>
</div>
</div>
<div class="section" id="transcriptome-index">
<h2>3) Transcriptome index<a class="headerlink" href="#transcriptome-index" title="Permalink to this headline">¶</a></h2>
<p>If you have a genome annotation in GTF format available, you should
create a transriptome index from that GTF file. Say you have an <tt class="docutils literal"><span class="pre">annotation.gtf</span></tt>,
file. You can create a transcriptome index like this:</p>
<div class="highlight-python"><pre>gemtools t-index -a annotation.gtf -i genome.gem -t 8</pre>
</div>
<p>The transcriptome indexer takes the annotation in GTF format and the previously
generated .gem index as input. The <tt class="docutils literal"><span class="pre">-t</span> <span class="pre">8</span></tt> again indicates that 8 threads should
be used by the process.</p>
<p>This creates 5 files in the current folder:</p>
<blockquote>
<div>annotation.gtf.junctions      &#8211; the splice junctions
annotation.gtf.junctions.fa   &#8211; the transcriptome
annotation.gtf.junctions.gem  &#8211; the transcriptome index
annotation.gtf.junctions.keys &#8211; keys to translate from transcriptome to genome
annotation.gtf.junctions.log  &#8211; indexer log file</div></blockquote>
<p>The pipeline needs all the files except the log file and its searching for them
automatically just next to the annotation fiel (if you do not specify the paths
explicitly). For a quick start, just keep the files next to the annotation.</p>
</div>
<div class="section" id="run-the-pipeline">
<h2>4) Run the pipeline<a class="headerlink" href="#run-the-pipeline" title="Permalink to this headline">¶</a></h2>
<p>With the index and the optional transcript index you can run the pipeline.
For this example, lets assume you have paired-end reads in two files
<tt class="docutils literal"><span class="pre">reads_1.fastq.gz</span></tt> and <tt class="docutils literal"><span class="pre">reads_2.fastq.gz</span></tt>.</p>
<p>Run this command to get an overview of what will happen:</p>
<div class="highlight-python"><pre>gemtools rna-pipeline -i genome.gem -a annotation.gtf -f reads_1.fastq -q 33 -t 8 --dry</pre>
</div>
<p>The tool will complain if anything is missing, otherwise it will print an
overview of the pipeline steps. Note that we assume here that everything is in
the same folder for the sake of simplicity. The main paramters are:</p>
<div class="highlight-python"><pre>-i genome.gem     -- the gem genome index
-a annotation.gtf -- the annotation. You can just skip this if you don't
                     have one. Otherwise the tool will search for the transcriptome
                     index just next to the annotation. If you put it
                     somewhere else you have to explicitly set the paths. Call
                     gemtools rna-pipeline --help for an overview of the available
                     options.

-f reads_1.fastq  -- the input file. NOTE that by default we assume you have
                     paired end reads and we search for a second file called
                     reads_2.fastq.gz (works also for other variants like
                     reads.0.fastq, reads_0.fq). If it doesn't find the
                     second file, specify it explicitly like
                     -f reads_1.fastq.gz reads_2.fastq.gz. If you do NOT
                     HAVE PAREID reads, just specify the input file and
                     add --single-end to the paramters.

-q 33             -- this is the quality offset. Should be 33 or 64, but
                     you have to specify this as figuring this out automatically
                     can be expensive.
-t 8              -- threads again. This is important as it significantly
                     speeds up the runs!</pre>
</div>
<p>Now, if everything looks good, start the same command but remove the <tt class="docutils literal"><span class="pre">--dry</span></tt>
to actually start the run. With the default configuration you will get these
output files:</p>
<div class="highlight-python"><pre>reads.map.gz    -- the gem aligned reads
reads.bam       -- alignments in bam format
reads.junctions -- the denovo junction sites found
reads.stats*    -- two sets of stats, *all* for all the mappings found,
                   *best* considering only the best mappings. You get two
                   files each. .txt is the human readable form, .json is
                   in JSON format so you can easily read the stats with
                   more or less any modern programming language.</pre>
</div>
</div>
<div class="section" id="run-a-stats-report-on-the-output">
<h2>5) Run a stats report on the output<a class="headerlink" href="#run-a-stats-report-on-the-output" title="Permalink to this headline">¶</a></h2>
<p>This step is optional, but allows a quick result check:</p>
<div class="highlight-python"><pre>gemtools report -i reads.stats.all.json -p</pre>
</div>
<p>The report command will create a .zip file with a graphical report (-p
indicates paired end reads). Unzip the file and open the index.html in
your web browser.</p>
</div>
<div class="section" id="notes">
<h2>Notes<a class="headerlink" href="#notes" title="Permalink to this headline">¶</a></h2>
<ul>
<li><p class="first">the pipeline is restartable. When it failed at some point you can just try to restart the run, it will skip the already completed steps.</p>
</li>
<li><p class="first">you can save pipeline configuration and reuse them later. This is handy if you have more datasets. For example:</p>
<div class="highlight-python"><pre>gemtools rna-pipeline -i genome.gem -a annotation.gtf -q 33 -t 8 --save config.gt</pre>
</div>
<p>will not run anything but save the configuration to a <tt class="docutils literal"><span class="pre">config.gt</span></tt> file.
You can reuse the configuration like this:</p>
<div class="highlight-python"><pre>gemtools rna-pipeline --load config.gt -f reads_A_1.fastq.gz
gemtools rna-pipeline --load config.gt -f reads_B_1.fastq.gz
gemtools rna-pipeline --load config.gt -f reads_C_1.fastq.gz
....</pre>
</div>
<p>All the configuration is taken from the config.gt file and you override the
input file from the command line. Note that the configuration file is stored in
JSON format. You can read this easily and also create configuration files
automatically.</p>
</li>
<li><p class="first"><tt class="docutils literal"><span class="pre">gemtools</span></tt> has a lot of options. Take a look with <tt class="docutils literal"><span class="pre">gemtools</span> <span class="pre">rna-pipeline</span> <span class="pre">--help</span></tt>. Here are a few important ones:</p>
<div class="highlight-python"><pre>--name         -- specify an output name, otherwise the input file name
                  is used as a template
--compress-all -- if you have very limited disk space, you can tell the
                  pipeline to compress all intermediate files on the fly.
                  This costs performance though!
--direct-input -- in case of limited disk space, add this to stream the
                  initial data directly into gem rather then creating a
                  dedicated input file.</pre>
</div>
</li>
</ul>
<p><em>LAST BUT VERY IMPORTANT NOTE</em></p>
<p>The pipeline expects to find <tt class="docutils literal"><span class="pre">samtools</span></tt> installed on the system. Try to get
the lates samtools from their github repository
(<a class="reference external" href="https://github.com/samtools/samtools">https://github.com/samtools/samtools</a> &#8211; clone or download and call make to
build it). The latest version is multi-threaded (i.e. <tt class="docutils literal"><span class="pre">samtools</span> <span class="pre">view</span> <span class="pre">--help</span></tt> will
show a <tt class="docutils literal"><span class="pre">-&#64;</span></tt> paramter). Also, see if you have <tt class="docutils literal"><span class="pre">pigz</span></tt> installed in the system
you try to run gemtools on. <tt class="docutils literal"><span class="pre">pigz</span></tt> is a parallel compressor and the pipeline
makes use of it if it is available. It will speed up compression steps a lot!</p>
</div>
<div class="section" id="running-into-problems">
<h2>Running into problems?<a class="headerlink" href="#running-into-problems" title="Permalink to this headline">¶</a></h2>
<p>Please do not hesitate to contact us if you run into any problems or you would
like to see other features implemented. Please consider using the <a class="reference external" href="https://github.com/gemtools/gemtools/issues?">GEMTools issue
tracker</a>.</p>
</div>
</div>
<div class="section" id="scoring-scheme">
<h1>Scoring Scheme<a class="headerlink" href="#scoring-scheme" title="Permalink to this headline">¶</a></h1>
<p>The <cite>rna-pipeline</cite> computes scores for each alignment. These scores are
transformed to the SAM MAPQ field in the resulting SAM file. Here is the basic
scheme, how the gem scores are calculated.</p>
<div class="section" id="best-strata">
<h2>Best strata<a class="headerlink" href="#best-strata" title="Permalink to this headline">¶</a></h2>
<p>The score for the best stratum is set as follows:</p>
<table border="1" class="docutils">
<colgroup>
<col width="53%" />
<col width="11%" />
<col width="18%" />
<col width="18%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Match class</th>
<th class="head">&nbsp;</th>
<th class="head">GEM score</th>
<th class="head">SAM score</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td rowspan="2">Unique, no subdominant match</td>
<td>Best</td>
<td>65504</td>
<td>254</td>
</tr>
<tr class="row-odd"><td>Worst</td>
<td>64512</td>
<td>252</td>
</tr>
<tr class="row-even"><td rowspan="2">Unique, no subdominant match(es)</td>
<td>Best</td>
<td>32736</td>
<td>180</td>
</tr>
<tr class="row-odd"><td>Worst</td>
<td>31744</td>
<td>177</td>
</tr>
<tr class="row-even"><td rowspan="2">Not unique, different score</td>
<td>Best</td>
<td>16352</td>
<td>127</td>
</tr>
<tr class="row-odd"><td>Worst</td>
<td>15360</td>
<td>123</td>
</tr>
<tr class="row-even"><td rowspan="2">Perfect tie</td>
<td>Best</td>
<td>14304</td>
<td>119</td>
</tr>
<tr class="row-odd"><td>Worst</td>
<td>13312</td>
<td>114</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="next-strata">
<h2>Next strata<a class="headerlink" href="#next-strata" title="Permalink to this headline">¶</a></h2>
<p>If the read contains more than one mapping, the score for the next stratum is
calculated as:</p>
<table border="1" class="docutils">
<colgroup>
<col width="53%" />
<col width="11%" />
<col width="18%" />
<col width="18%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Match class</th>
<th class="head">&nbsp;</th>
<th class="head">GEM score</th>
<th class="head">SAM score</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td rowspan="2">Unique, no subdominant match</td>
<td>Best</td>
<td>8160</td>
<td>89</td>
</tr>
<tr class="row-odd"><td>Worst</td>
<td>5120</td>
<td>71</td>
</tr>
<tr class="row-even"><td rowspan="2">Unique, no subdominant match(es)</td>
<td>Best</td>
<td>12256</td>
<td>110</td>
</tr>
<tr class="row-odd"><td>Worst</td>
<td>9216</td>
<td>95</td>
</tr>
<tr class="row-even"><td rowspan="2">Not unique, different score</td>
<td>Best</td>
<td>12256</td>
<td>110</td>
</tr>
<tr class="row-odd"><td>Worst</td>
<td>9216</td>
<td>95</td>
</tr>
<tr class="row-even"><td rowspan="2">Perfect tie</td>
<td>Best</td>
<td>12256</td>
<td>110</td>
</tr>
<tr class="row-odd"><td>Worst</td>
<td>9216</td>
<td>95</td>
</tr>
</tbody>
</table>
<p>In addition, the optional SAM field <em>XT</em> is set to <em>R</em> for all perfect ties and to <em>U</em> for all other alignments.</p>
</div>
</div>


      </div>
      <div class="bottomnav">
      
        <p>
        «&#160;&#160;<a href="index.html">Welcome to GEMTools&#8217;s documentation!</a>
        &#160;&#160;::&#160;&#160;
        <a class="uplink" href="index.html">Contents</a>
        </p>

      </div>

    <div class="footer">
        &copy; Copyright 2013, Santiago Marco-Sola, Thasso Griebel.
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.1.3.
    </div>
  </body>
</html>